### PR TITLE
[#MHV-47421] Get a patient's MHV FHIR ID from their MHV correlation ID

### PR DESCRIPTION
**Pull request by mmoyer-va**
## Summary

In order to use the MHV FHIR-ized APIs, we need to pass the user's Patient FHIR ID in each call. To that end, when a user initializes their MR session, we now get that ID in three steps:
1. MHV correlation ID is passed into the session.
1. With that ID, an authentication call is made, returning a JWT. The JWT is parsed and the subject ID is extracted.
1. A call is made to the FHIR Patient endpoint with the subject ID, and the patient's FHIR ID is extracted. It is stored in session for use in all further calls.

## Related issue(s)
- [MHV-47421](https://jira.devops.va.gov/browse/MHV-47421) - MR to VA.gov: Tie the authenticated user to a FHIR ID in their MHV MR Session

## Testing done

- New test was written to ensure the Patient endpoint works properly
- Existing client authentication tests were modified to ensure the FHIR ID is being pulled and passed to all calls

## What areas of the site does it impact?
- MHV Medical Records application

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by mattwrightva — APPROVED** [review]
Looks good!

**Comment by Wyzardsleeves** [comment]
Look good!

**Review by Wyzardsleeves — APPROVED** [review]
okay!

File: lib/common/client/concerns/mhv_jwt_session_client.rb [hunk 1]
```diff
@@ -46,9 +46,20 @@ def get_session
           env = get_session_tagged
           # req_headers = env.request_headers
           res_headers = env.response_headers
+
+          # Decode the JWT token and extract the user_id from the payload
           jwt_token = res_headers['authorization'].sub('Bearer ', '')
+          decoded_token = JWT.decode jwt_token, nil, false
+          subject_id = decoded_token[0]['subjectId']
```
Comment (rmtolmach) [inline lib/common/client/concerns/mhv_jwt_session_client.rb:53]: Can you be sure `decoded_token` will always have an index `0` and that it has a `subjectId`, or does there need to be any error handling here?

Same question for line 58 in this file.

Comment (mmoyer-va) [reply lib/common/client/concerns/mhv_jwt_session_client.rb]: @rmtolmach that's a great point. I beefed up the error handling in that function.

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**